### ISSUE TITLE
Mouse up outside form dialog while selecting text inside a form field closes the form

### ISSUE THREAD

**Issue by neilyoung** [ant-design/ant-design#17628]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://dragonfly-demo.accuware.com:8443](https://dragonfly-demo.accuware.com:8443)

### Steps to reproduce
Click "Join"
Enter text in "Room name"
Mark it (e.g. for copying), move mouse outside the form window
Release mouse

### What is expected?
Form should not close

### What is actually happening?
Form is closing

| Environment | Info |
|---|---|
| antd | 3.20.2 |
| React | 16.4.1 |
| System | macOS 10.14.5 |
| Browser | Chrome 75.0 |

**Comment by neilyoung** [comment]
Disregard. Looks like my bad

**Comment by neilyoung** [comment]
Well, no. There is some strange issue. In my form onCancel is triggered. This is not the case in your codepen sample of the form description page.

**Comment by neilyoung** [comment]
This is my form:

```
import React from 'react'
import { Icon, Modal, Form, Input } from 'antd'


const RoomCreateForm = Form.create({ name: 'room-form' })(

  class extends React.Component {


    render() {
      const { visible, onCancel, onCreate, form } = this.props
      const { getFieldDecorator } = form

      return (
        <Modal
          visible={visible}
          title="Join Conference Room"
          okText="Enter"
          onCancel={onCancel}
          onOk={onCreate}
        >
          <Form layout="vertical">
            <Form.Item label="Room name" >
              {getFieldDecorator('roomName', {
                initialValue: this.props.room.roomName,
                rules: [{ required: true, message: 'Please enter the name of the room' }],
              })(
                <Input prefix={<Icon type="home"  />} placeholder="Room which should be joined" autoFocus />
              )}
            </Form.Item>

            <Form.Item label="Your name in the room">
              {getFieldDecorator('userName', { initialValue: this.props.room.userName })(
                <Input prefix={<Icon type="user" />} placeholder="Your published name in the room" />
              )}
            </Form.Item>

          </Form>
        </Modal>
      )
    }
  }
)

export default RoomCreateForm
```

**Comment by zombieJ** [comment]
hi @neilyoung , pls help to create an online reproduce with code: https://u.ant.design/codesandbox

**Comment by neilyoung** [comment]
Figured it out. It was obviously something which was fixed in your code between 3.15.2 and 3.20.3.
